site/docs/snapshot-tag-branch.md

```diff
@@ -0,0 +1,148 @@
+Retention policy can be configured both globally and on snapshot reference to provide highly flexible customization to the expiration behavior.
+
+#### Global snapshot retention policy
+
+Global snapshot retention policy can be set through the following table properties:
+
+| Property                             | Default            | Description                                                   |
+| ------------------------------------ | ------------------ | ------------------------------------------------------------- |
+| history.expire.max-snapshot-age-ms   | 432000000 (5 days) | Default max age of snapshots to keep while expiring snapshots    |
+| history.expire.min-snapshots-to-keep | 1                  | Default min number of snapshots to keep while expiring snapshots |
+
+#### Snapshot reference retention policy
+
+Similarly, snapshot reference has the properties below to provider finer grain control:
+
+| Property                     | Type      | Description |
+|------------------------------|-----------|-------------|
+| **`min-snapshots-to-keep`**  | `int`     | For `branch` type only, the minimum number of snapshots to keep in a branch |
+| **`max-snapshot-age-ms`**    | `long`    | The duration before a snapshot tagged or in a branch could be expired by any automatic snapshot expiration process  |
+
+#### Policy evaluation mechanism
+
+When a snapshot expiration process starts, it follows the steps described below:
+
+1. form an expiration candidate pool containing all snapshots
+2. for each snapshot reference, evaluate the associated policy and move snapshots out of the candidate pool
+3. apply global retention policy to and move snapshots out of the candidate pool
+4. when multiple snapshots can be chosen to be moved out, newer snapshots win
+4. after evaluation, expire all snapshots that are still in the candidate pool
+
+#### Policy evaluation example
+
+Here is an example for how an Iceberg snapshot expiration procedure evaluates what snapshots to expire.
+
+Suppose we have the following snapshot graph and retention policies configured:
+
+```
+A -> B -> C (main)
+ \      (dev)
+  D -> E (b1)
+   \
+    F -> G (b2)
+```
+
+| Policy Type      | Max Age       | Min to Keep | Snapshots Affected    |
+|------------------|---------------|-------------|-----------------------|
+| global           | 5 hours       | 4           | A, B, C, D, E, F, G   |
+| branch/main      | 3 days        | 1           | A, B, C               |
+| branch/b1        | 2 days        | 2           | A, D,  E              |
+| branch/b2        | 1 day         | 0           | A, D, F, G            |
+| tag/dev          | forever       | N/A         | C                     |
+
+Assume that we have a process continuously running the snapshot expiration procedure, we would have the results below as time progresses:
+
+##### Day 1: F and G are expired
+
+On day 1, the global and branch b2 max age has passed, affecting A, D, F, G.  
```

Comment:
Is it better to have some snapshots in the example that are not referenced by branches and tags just to apply global retention policy ? now all snapshots are under branch or tag. so global policy cannot be applied. (This is based on my assumption, please see last comment for this page)

Reply:
sure, I will try to add one